lib/Monads/trace/Trace_RG.thy

```diff
@@ -108,11 +192,11 @@ lemma guar_cond_append:
   by (simp add: guar_cond_def trace_steps_append ball_Un conj_comms)
 
 lemma prefix_closed_bind:
-  "prefix_closed f \<Longrightarrow> (\<forall>x. prefix_closed (g x)) \<Longrightarrow> prefix_closed (f >>= g)"
+  "\<lbrakk>prefix_closed f; \<forall>x. prefix_closed (g x)\<rbrakk> \<Longrightarrow> prefix_closed (f >>= g)"
   apply (subst prefix_closed_def, clarsimp simp: bind_def)
   apply (auto simp: Cons_eq_append_conv split: tmres.split_asm
              dest!: prefix_closedD[rotated];
-    fastforce elim: rev_bexI)
+         fastforce elim: rev_bexI)
```

Comment:
this indent doesn't seem right

Reply:
The indent seems right to me, but the `auto ...; fastforce` method is obviously terrible. I'll see if I can rework it into something nicer.

Reply:
The `fastforce` is to the left of the `(`, which can't be right. The `dest!:` on the `auto` should align with the `simp`, not the `auto`, and then the `fastforce` goes after the semicolon, so should be on the same level as the `auto`.
That said, I agree with the method itself being terrible, so either approach works.

Reply:
Also happy to leave this one for later, yes.